### PR TITLE
Delete eligibility session data on start step

### PR DESCRIPTION
**Pull request by frederike-ramin**
# Short Description
- The marital status (and all other data) is now deleted when you restart the eligibility flow as this was before irritating.

# Changes
- Move `override_session_data` to super class to be able to explicitly call it in `DisplayStep`s if needed
- Call `override_session_data` in the eligibility start step to remove all pre-existing data

# Feedback
- Do you think the tests cover what I did there?

### CODE REVIEW

File: webapp/app/forms/steps/eligibility_steps.py [hunk 1]
```diff
@@ -189,7 +189,10 @@ def __init__(self, **kwargs):
             **kwargs)
 
     def _main_handle(self, stored_data):
-        stored_data = super()._main_handle(stored_data)
+        super()._main_handle(stored_data)
```
Comment (JulianRoesner) [inline webapp/app/forms/steps/eligibility_steps.py:192]: I know, that we do not need it here. However, imo we could still set it here as always just to ensure that we always do it like that

File: webapp/tests/forms/steps/test_eligibility_steps.py [hunk 2]
```diff
@@ -175,6 +176,36 @@ def test_if_invalid_and_valid_and_second_valid_are_in_prev_step_then_set_prev_ur
...
+
+            self.assertEqual({}, deserialize_session_data(req.session[_ELIGIBILITY_DATA_KEY]))
+
+    def test_does_not_change_other_session_data(self):
+        other_session_key = 'OTHER_SESSION_KEY'
+        other_session_data = {'Galileo': 'Figaro - magnificoo'}
+        session_data = {
+            _ELIGIBILITY_DATA_KEY: create_session_form_data({'marital_status_eligibility': 'single'}),
+            other_session_key: create_session_form_data(other_session_data)
+        }
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/test_eligibility_steps.py:198]:
```suggestion
        other_session_key = 'OTHER_SESSION_KEY'
        other_session_data ={ 'OTHER_SESSION_KEY': {'Galileo': 'Figaro - magnificoo'}, 'ANOTHER_SESSION_KEY':  {'Scaramouch': 'Fandango'}}
        session_data = {_ELIGIBILITY_DATA_KEY: create_session_form_data({'marital_status_eligibility': 'single'})} | other_session_data
```
Just a suggestion to ensure that more than one other key is kept. And then we could just assert that the data is other_session_data below.

File: webapp/tests/forms/steps/test_eligibility_steps.py [hunk 3]
```diff
@@ -175,6 +176,36 @@ def test_if_invalid_and_valid_and_second_valid_are_in_prev_step_then_set_prev_ur
...
+            req.session = SecureCookieSession(session_data)
+            step = EligibilityStepChooser('eligibility').get_correct_step(
+                EligibilityStartDisplaySteuerlotseStep.name)
+            step.handle()
+
+            self.assertEqual({}, deserialize_session_data(req.session[_ELIGIBILITY_DATA_KEY]))
+
+    def test_does_not_change_other_session_data(self):
+        other_session_key = 'OTHER_SESSION_KEY'
+        other_session_data = {'Galileo': 'Figaro - magnificoo'}
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/test_eligibility_steps.py:194]: ❤️

File: webapp/tests/forms/steps/test_eligibility_steps.py [hunk 4]
```diff
@@ -175,6 +176,36 @@ def test_if_invalid_and_valid_and_second_valid_are_in_prev_step_then_set_prev_ur
             self.assertEqual(previous_url, step.render_info.prev_url)
 
 
+class TestEligibilityStartDisplaySteuerlotseStep(unittest.TestCase):
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/test_eligibility_steps.py:179]: Personally, I would add one additional tests for a totally empty session_data and one for empty eligibility data. Nevertheless, this looks good to me.

**Review by JulianRoesner — APPROVED** [review]
Looks very good! I think that you should move the test in `TestSteuerlotseFormStepOverrideSessionData` to `TestSteuerlotseStepOverrideSessionData` and test them for the SteuerlotseStep as we've moved the method?

File: webapp/tests/forms/steps/test_eligibility_steps.py (hunk 2 above)
Comment (frederike-ramin) [reply webapp/tests/forms/steps/test_eligibility_steps.py:198]: Unfortunately, this way, the other session data is not encrypted. But I'll add an additional key.

File: webapp/tests/forms/steps/test_eligibility_steps.py (hunk 4 above)
Comment (frederike-ramin) [reply webapp/tests/forms/steps/test_eligibility_steps.py:179]: Added them. Is that what you had in mind?

Comment (JulianRoesner) [reply webapp/tests/forms/steps/test_eligibility_steps.py:153]: Yep
